### PR TITLE
patch: support multiple patch paths or dirs

### PR DESCRIPTION
**Pull request by Kybxd**
relevant to https://github.com/tableauio/tableau/pull/156

### CODE REVIEW

File: cmd/protoc-gen-cpp-tableau-loader/embed/hub.pc.h [hunk 1]
```diff
@@ -52,11 +52,11 @@ struct LoadOptions {
...
-  // Patch paths maps each messager name to a corresponding patch file path.
-  // If specified, then main messager will patched.
-  std::unordered_map<std::string, std::string> patch_paths;
-  // Specify the patch directory for config patching.
-  std::string patch_dir;
+  // PatchPaths maps each messager name to one or multiple corresponding patch file paths.
+  // If specified, then main messager will be patched.
+  std::unordered_map<std::string, std::vector<std::string>> patch_paths;
+  // PatchDirs specifies the directory paths for config patching.
+  std::vector<std::string> patch_dirs;
```
Comment (wenchy) [inline cmd/protoc-gen-cpp-tableau-loader/embed/hub.pc.h:59]: `PatchPaths` -> `Patch paths`
`PatchDirs` -> `Patch dirs`

File: cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h [hunk 2]
```diff
@@ -84,3 +101,9 @@ void SetDefaultLogger(Logger* logger);
 #define ATOM_ERROR(...) ATOM_LOGGER_CALL(tableau::log::DefaultLogger(), tableau::log::kError, __VA_ARGS__)
 #define ATOM_FATAL(...) ATOM_LOGGER_CALL(tableau::log::DefaultLogger(), tableau::log::kFatal, __VA_ARGS__)
 
+#define ATOM_PRINT_VECTOR(v) \
+  tableau::log::DebugString(v, [](std::stringstream& ss, decltype(v)::const_iterator it) { ss << *it; })
+#define ATOM_PRINT_MAP(m)                                                                  \
```
Comment (wenchy) [inline cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h:106]: `ATOM_PRINT_VECTOR` -> `ATOM_VECTOR_STR`
`ATOM_PRINT_MAP` -> `ATOM_MAP_STR`

File: cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h [hunk 3]
```diff
@@ -70,6 +70,23 @@ const char* NowStr();
 Logger* DefaultLogger();
 void SetDefaultLogger(Logger* logger);
 
+template <typename DataStructure>
```
Comment (wenchy) [inline cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h:73]: `DataStructure` -> `T`

File: cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h [hunk 4]
```diff
@@ -84,3 +101,9 @@ void SetDefaultLogger(Logger* logger);
 #define ATOM_ERROR(...) ATOM_LOGGER_CALL(tableau::log::DefaultLogger(), tableau::log::kError, __VA_ARGS__)
 #define ATOM_FATAL(...) ATOM_LOGGER_CALL(tableau::log::DefaultLogger(), tableau::log::kFatal, __VA_ARGS__)
 
+#define ATOM_PRINT_VECTOR(v) \
+  tableau::log::DebugString(v, [](std::stringstream& ss, decltype(v)::const_iterator it) { ss << *it; })
+#define ATOM_PRINT_MAP(m)                                                                  \
+  tableau::log::DebugString(m, [](std::stringstream& ss, decltype(m)::const_iterator it) { \
+    ss << "{" << it->first << ", " << it->second << "}";                                   \
```
Comment (wenchy) [inline cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h:108]: `,` -> `: `

**Review by wenchy — CHANGES_REQUESTED** [review]

File: cmd/protoc-gen-cpp-tableau-loader/embed/hub.pc.h (hunk 1 above)
Comment (Kybxd) [reply cmd/protoc-gen-cpp-tableau-loader/embed/hub.pc.h]: done

File: cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h (hunk 2 above)
Comment (Kybxd) [reply cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h]: done

File: cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h (hunk 3 above)
Comment (Kybxd) [reply cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h]: done

File: cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h (hunk 4 above)
Comment (Kybxd) [reply cmd/protoc-gen-cpp-tableau-loader/embed/logger.pc.h]: done

**Review by wenchy — APPROVED** [review]

**Pull request merged**